### PR TITLE
Adding Multitasking View to Super Key Options

### PR DESCRIPTION
**Pull request by MobileAZN**
Adding the ability for the Super key to also show the Multitasking View.

### CODE REVIEW

File: src/Views/Layout.vala [hunk 1]
```diff
@@ -212,9 +214,12 @@ namespace Pantheon.Keyboard {
                 case "io.elementary.wingpanel --toggle-indicator=app-launcher":
                     overlay_key_combo.active = 1;
                     break;
-                case "io.elementary.shortcut-overlay":
+                case "show-desktop":
```
Comment (davidmhewitt) [inline src/Views/Layout.vala:217]:
```suggestion
                case "dbus-send --session --dest=org.pantheon.gala --print-reply /org/pantheon/gala org.pantheon.gala.PerformAction int32:1":
```

This string needs to match the command you use further down. It's responsible for reading the GSetting and selecting the correct item in the combobox (i.e. on startup). So you'll probably find that your new Multitasking View item is not automatically selected when you restart switchboard after selecting it.

**Review by davidmhewitt — CHANGES_REQUESTED** [review]
There's a minor issue with the current code (I've left a comment inline), then it'll need a review from the UX team to decide if this is a feature that should be implemented in this way.

File: src/Views/Layout.vala (hunk 1 above)
Comment (MobileAZN) [reply src/Views/Layout.vala]: Updated line 217 to match.

**Comment by cassidyjames** [comment]
UX-wise I think it makes sense to offer the multitasking view as an option for Super, especially because I've heard the desire from GNOME converts. I'll leave the code review to @elementary/desktop-developers though.

**Review by cassidyjames — COMMENTED** [review]
UX-wise I think it makes sense to offer the multitasking view as an option for Super, especially because I've heard the desire from GNOME converts. I'll leave the code review to @elementary/desktop-developers though.

**Comment by jeremypw** [comment]
I do not like that a long and complicated string is written out explicitly twice, although a similar style is used in existing code.

**Review by davidmhewitt — CHANGES_REQUESTED** [review]
> I do not like that a long and complicated string is written out explicitly twice, although a similar style is used in existing code.

Yeah, I'm inclined to agree, can we have the command string brought out into a constant?

Something like `private const string MULTITASKING_COMMAND = "..."` at the top of the class, and then use that constant instead of writing it out twice.

**Comment by MobileAZN** [comment]
I have updated this to the suggested constant and cleaner call.

File: src/Views/Layout.vala [hunk 2]
```diff
@@ -24,6 +24,7 @@ namespace Pantheon.Keyboard {
         private Gtk.SizeGroup [] size_group;
         private AdvancedSettings advanced_settings;
         private Gtk.Entry entry_test;
+        private const string MultitaskingView = "dbus-send --session --dest=org.pantheon.gala --print-reply /org/pantheon/gala org.pantheon.gala.PerformAction int32:1";
```
Comment (davidmhewitt) [inline src/Views/Layout.vala:27]:
```suggestion
        private const string MULTITASKING_VIEW_COMMAND = "dbus-send --session --dest=org.pantheon.gala --print-reply /org/pantheon/gala org.pantheon.gala.PerformAction int32:1";
```

File: src/Views/Layout.vala [hunk 3]
```diff
@@ -212,9 +215,12 @@ namespace Pantheon.Keyboard {
                 case "io.elementary.wingpanel --toggle-indicator=app-launcher":
                     overlay_key_combo.active = 1;
                     break;
-                case "io.elementary.shortcut-overlay":
+                case MultitaskingView:
```
Comment (davidmhewitt) [inline src/Views/Layout.vala:218]:
```suggestion
                case MULTITASKING_VIEW_COMMAND:
```

File: src/Views/Layout.vala [hunk 4]
```diff
@@ -238,6 +244,8 @@ namespace Pantheon.Keyboard {
                 } else if (combo_active == 1) {
                     gala_behavior_settings.set_string ("overlay-action", "io.elementary.wingpanel --toggle-indicator=app-launcher");
                 } else if (combo_active == 2) {
+                    gala_behavior_settings.set_string ("overlay-action", MultitaskingView);
```
Comment (davidmhewitt) [inline src/Views/Layout.vala:247]:
```suggestion
                    gala_behavior_settings.set_string ("overlay-action", MULTITASKING_VIEW_COMMAND);
```

**Review by davidmhewitt — CHANGES_REQUESTED** [review]
elementary code style is to use `UPPERCASE_UNDERSCORE_NAMING` for constants, so I've left those changes as a suggestion.

**Comment by MobileAZN** [comment]
ah, ok.  Updated using the sane defaults you offered @davidmhewitt

**Comment by davidmhewitt** [comment]
This is looking good now, but I'm going to mark it as blocked until we get https://github.com/elementary/gala/pull/1393 merged. I don't think this option should be available until it's also possible for the multitasking view to be closed with another press of the super key. So we'll await the status of that PR.

**Review by davidmhewitt — APPROVED** [review]
Thanks for your work on this!

**Comment by MobileAZN** [comment]
haha!  That was my next question, is where do I look to be able to close it with the same key vs hitting esc, as that was the next thing I wanted to look at.  Thanks again for the help!

**Pull request merged**